fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -535,6 +539,19 @@ public void setBaseTableInfos(List<BaseTableInfo> baseTableInfos) {
         this.baseTableInfos = baseTableInfos;
     }
 
+    public List<TableType> getBaseTableTypes() {
+        if (baseTableTypes == null) {
```

Comment:
can we just use ImmutableList for baseTableTypes and initialized to be null, then
```java
if (baseTableTypes==null) {
    baseTableTypes = baseTableInfos.stream()
            .map(BaseTableInfo::getTable)
            .map(Table::getType)
            .collect(Immutablelist.toImmutableList());
}
return baseTableTypes;
```

Reply:
removed. fixed.